qiskit/pulse/reschedule.py

```diff
@@ -141,27 +141,21 @@ def add_implicit_acquires(schedule: ScheduleComponent, meas_map: List[List[int]]
     """
     new_schedule = Schedule(name=schedule.name)
 
+    acquire_instructions = {}
     for time, inst in schedule.instructions:
         if isinstance(inst, AcquireInstruction):
-            if any([acq.index != mem.index for acq, mem in zip(inst.acquires, inst.mem_slots)]):
-                warnings.warn("One of your acquires was mapped to a memory slot which didn't match"
-                              " the qubit index. I'm relabeling them to match.")
-            cmd = Acquire(inst.duration, inst.command.discriminator, inst.command.kernel)
-            # Get the label of all qubits that are measured with the qubit(s) in this instruction
-            existing_qubits = {chan.index for chan in inst.acquires}
-            all_qubits = []
             for sublist in meas_map:
-                if existing_qubits.intersection(set(sublist)):
-                    all_qubits.extend(sublist)
-            # Replace the old acquire instruction by a new one explicitly acquiring all qubits in
-            # the measurement group.
-            new_schedule |= AcquireInstruction(
-                cmd,
-                [AcquireChannel(i) for i in all_qubits],
-                [MemorySlot(i) for i in all_qubits]) << time
+                for acq in inst.acquires:
+                    if acq.index in set(sublist):
+                        for i in sublist:
+                            acquire_instructions[i] = (time, inst)
```

Comment:
What if qubit 0 is acquired twice? (The hardware might not support it now, but we should try to be ahead of the hardware)

Reply:
hm, should we warn user that last instruction will be applied?

Reply:
looks like https://github.com/Qiskit/qiskit-terra/pull/3574/commits/f7d6249753cdf8c8afb9f7c148db9a5c61a65229 should overcome that issue.

P.S.
Funny enough that there are no usages of this method except tests :)